modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -22,6 +22,10 @@ const (
 	metricSizeTotal       = "traces_spanmetrics_size_total"
 )
 
+var (
+	intrinsicDimensions = []string{"service", "span_name", "span_kind", "status_code", "status_message"}
```

Comment:
seeing the way the dimensions are added in `New()` and knowing that order is important this may be less valuable than I think.

my instinct is this change is not worth it. wdyt?

Reply:
I agree that the performance gain should be negligible, especially since the lookup only occurs in `New()`. Therefore, I don't think it's work it.

However, I was curious and made a micro benchmark comparing string lookup in small slices and maps: both are pretty much comparable for slices and maps with less than 10 elements. I think this supports your instinct.

Reply:
My bad. I misread that. I thought that `sanitizeLabelNameWithCollisions` was called for every processed span. Agree just in `New()` this way is better.